src/Core/Scheduler/Providers/PointCloudProvider.js

```diff
@@ -130,6 +130,51 @@ function loadPointFile(layer, url) {
     });
 }
 
+function pick(view, mouse, layer) {
```

Comment:
Good refacto and unification.
However I have an objection on the choice of files for the picking declaration : the providers 

To be more generic, clearer and to reuse,  should be in a picking toolbox.
These tools must have as input the type of object to pick.

Then the layer's picking functionality must be assigned to its construction, but not in the pre-process phase of the provider. Because the provider must take care to supply only what has in title and pre-process works only for this supply.

it's same case in PR #546 
what is your motivation to put this in the providers?

I also ask the opinion to  @mbredif ?

Reply:
> To be more generic, clearer and to reuse, should be in a picking toolbox.
> [...]
>  what is your motivation to put this in the providers?

Yes; we already had this discussion earlier, so I won't copy-paste all my arguments from there.

I can move `pickObjectsAt` to a `Picking.js` file if it you prefer.

But I still believe that `preprocessing` of geometry Providers is a good place to extend layer's object so I'll leave the `layer.pickObjectsAt` there.

Reply:
Done.